stix_shifter_modules/gcp_chronicle/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,579 @@
+import copy
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, \
+    ComparisonExpression, ComparisonComparators, Pattern, CombinedComparisonExpression, CombinedObservationExpression
+import logging
+import re
+from datetime import datetime, timedelta
+from os import path
+import json
+
+logger = logging.getLogger(__name__)
+
+START_STOP_PATTERN = r"(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z)"
+MAC = '^(([0-9a-fA-F]{2}[:-]){5}([0-9a-fA-F]{2}))$'
+EMAIL = r'^\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b$'
+CONFIG_MAP_PATH = "json/config_map.json"
+
+STOP_TIME = datetime.utcnow()
+
+
+class StartStopQualifierValueException(Exception):
+    pass
+
+
+class FileNotFoundException(Exception):
+    pass
+
+
+class QueryStringPatternTranslator:
+    """
+    translate stix pattern to native data source query language
+    """
+
+    def __init__(self, pattern: Pattern, data_model_mapper, options):
+        logger.info("GCP Chronicle Connector")
+        self.dmm = data_model_mapper
+        self.options = options
+        self.all_queries = []
+        self.comparator_lookup = self.dmm.map_comparator()
+        self.config_map = self.load_json(CONFIG_MAP_PATH)
+        self.parse_expression(pattern)
+        self.qualified_query = self._create_formatted_query()
+
+    @staticmethod
+    def load_json(rel_path_of_file):
+        """
+        Consumes a json file and returns a dictionary
+        :param rel_path_of_file: str
+        :return: dict
+        """
+
+        _json_path = path.dirname(path.realpath(__file__)) + "/" + rel_path_of_file
```

Comment:
read from absolute path.

Reply:
Updated realpath to abspath as suggested.